src/python/pants/util/resources.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+
+import importlib
+from itertools import chain
+
+import importlib_resources as resources
+
+
+def read_resource(package_or_module: str, resource: str) -> bytes:
+    """Reads a resource file from within the Pants package itself.
+
+    This helper function is designed for compatibility with `pkgutil.get_data()` wherever possible,
+    but also allows compability with PEP302 pluggable importers such as included with PyOxidizer.
+    This requires that resources are loaded from a valid Python package (i.e. must have an
+    `__init__.py` file in the directory).
+    """
+
+    a = importlib.import_module(package_or_module)
+    package_ = a.__package__
+
+    if package_ is None:
+        raise ValueError(
+            "`read_resource` can only help find resources for packages or modules that live in "
+            "a package."
+        )
+
+    resource_parts = resource.split("/")
+
+    if len(resource_parts) == 1:
+        package = package_
+    else:
+        package = ".".join(chain((package_,), resource_parts[:-1]))
+        resource = resource_parts[-1]
+
+    return resources.read_binary(package, resource)
```

Comment:
It looks like Pants production code does not use this but:
```
$ tree foo/
foo/
├── bar
│   └── baz
│       └── spam.txt
└── __init__.py

2 directories, 2 files
```
```python
$ python3.9
Python 3.9.13 (main, May 29 2022, 10:29:51) 
[GCC 12.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from importlib import resources
>>> import pkgutil
>>> pkgutil.get_data("foo", "bar/baz/spam.txt")
b''
>>> resources.read_binary("foo.bar.baz", "spam.txt")
Traceback (most recent call last):
  File "/home/jsirois/.pyenv/versions/3.9.13/lib/python3.9/importlib/resources.py", line 97, in open_binary
    return open(full_path, mode='rb')
FileNotFoundError: [Errno 2] No such file or directory: '/home/jsirois/dev/pantsbuild/jsirois-pex/spam.txt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jsirois/.pyenv/versions/3.9.13/lib/python3.9/importlib/resources.py", line 126, in read_binary
    with open_binary(package, resource) as fp:
  File "/home/jsirois/.pyenv/versions/3.9.13/lib/python3.9/importlib/resources.py", line 111, in open_binary
    raise FileNotFoundError(message)
FileNotFoundError: 'spam.txt' resource not found in 'foo.bar.baz'
>>> 
```
Perhaps just letting resources.read_binary fail is enough, but it may make sense to note this restriction in the function doc.